### PR TITLE
Data Picker is too big

### LINKED ISSUE #41182: Data Picker is too big

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #41182

### Description

See [Slack](https://metaboat.slack.com/archives/C0645JP1W81/p1712586211035699) for context.

### How to verify

1. Start a new GUI question/model or edit an existing one
2. Choose data source

### Demo

[before](https://github.com/metabase/metabase/assets/6830683/b42d0ead-32ca-41fe-a976-8d9e793972c0) & [after](https://github.com/metabase/metabase/assets/6830683/2bfd6aa0-95ce-4e73-9f9e-7f02cd5f1673)

### CODE REVIEW

File: frontend/src/metabase/common/components/DataPicker/components/TablePicker.module.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.inner {
+  padding: 10dvh 10dvw !important;
```
Comment (uladzimirdev) [inline frontend/src/metabase/common/components/DataPicker/components/TablePicker.module.css:2]: I don't like using `!important` tbh @ranquild seems it's going to be a common pattern

File: frontend/src/metabase/common/components/DataPicker/components/DataPickerModal.tsx [hunk 2]
```diff
@@ -46,6 +47,9 @@ export const DataPickerModal = ({ value, onChange, onClose }: Props) => {
   return (
     <EntityPickerModal
       canSelectItem
+      classNames={{
```
Comment (uladzimirdev) [inline frontend/src/metabase/common/components/DataPicker/components/DataPickerModal.tsx:50]: maybe `modalClassNames`?

**Review by uladzimirdev — APPROVED** [review]

**Comment by ranquild** [comment]
We should't hack components like this. Please discuss with the design team. If needed, apply a fix / add a variant etc to the component.

File: frontend/src/metabase/common/components/DataPicker/components/TablePicker.module.css (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/common/components/DataPicker/components/TablePicker.module.css:2]: Replied

Comment (ranquild) [reply frontend/src/metabase/common/components/DataPicker/components/TablePicker.module.css:2]: Short answer - don't add hacks

File: frontend/src/metabase/common/components/EntityPicker/components/EntityPickerModal/EntityPickerModal.tsx [hunk 3]
```diff
@@ -95,6 +98,7 @@ export function EntityPickerModal<
 
   return (
     <Modal.Root
+      classNames={classNames}
```
Comment (ranquild) [inline frontend/src/metabase/common/components/EntityPicker/components/EntityPickerModal/EntityPickerModal.tsx:101]: Have you tried passing `padding` or `p` here or on subcomponents?

File: frontend/src/metabase/common/components/EntityPicker/components/EntityPickerModal/EntityPickerModal.tsx [hunk 4]
```diff
@@ -47,6 +47,7 @@ export interface EntityPickerModalProps<Model extends string, Item> {
   searchResultFilter?: (results: Item[]) => Item[];
   actionButtons?: JSX.Element[];
   trapFocus?: boolean;
+  variant?: "large-overlay-padding";
```
Comment (ranquild) [inline frontend/src/metabase/common/components/EntityPicker/components/EntityPickerModal/EntityPickerModal.tsx:50]: Maybe it should be in all cases for `EntityPickerModal`?

File: frontend/src/metabase/ui/components/overlays/Modal/Modal.styled.tsx [hunk 5]
```diff
@@ -16,6 +16,13 @@ export const getModalOverrides = (): MantineThemeOverride["components"] => ({
         backgroundColor: theme.fn.rgba(theme.fn.themeColor("bg-black"), 0.6),
       },
     }),
+    variants: {
+      "large-overlay-padding": () => ({
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/overlays/Modal/Modal.styled.tsx:20]: I'd pick a semantic name if possible. Not sure if there is one. Someone from the design team could help

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase/common/components/EntityPicker/components/EntityPickerModal/EntityPickerModal.tsx (hunk 4 above)
Comment (kamilmielnik) [reply frontend/src/metabase/common/components/EntityPicker/components/EntityPickerModal/EntityPickerModal.tsx:50]: Asked in [Slack](https://metaboat.slack.com/archives/C0645JP1W81/p1712674703445559?thread_ts=1712586211.035699&cid=C0645JP1W81).

File: frontend/src/metabase/ui/components/overlays/Modal/Modal.styled.tsx (hunk 5 above)
Comment (uladzimirdev) [reply frontend/src/metabase/ui/components/overlays/Modal/Modal.styled.tsx:20]: agreed, we need a better name for the variant

**Review by mazameli — APPROVED** [review]

**Review by uladzimirdev — APPROVED** [review]

**Review by uladzimirdev — APPROVED** [review]

**Pull request merged**